sky/utils/schemas.py

```diff
@@ -370,6 +375,70 @@ def get_service_schema():
     }
 
 
+def _filter_schema(schema: dict, keys_to_keep: dict) -> dict:
+    """Recursively filter a schema to include only certain keys.
+
+    Args:
+        schema: The original schema dictionary.
+        keys_to_keep: List of tuples with the path of keys to retain.
+
+    Returns:
+        The filtered schema.
+    """
+    # Convert list of tuples to a dictionary for easier access
+    paths_dict: Dict[str, Any] = {}
+    for path in keys_to_keep:
+        current = paths_dict
+        for step in path:
+            if step not in current:
+                current[step] = {}
+            current = current[step]
+
+    def keep_keys(current_schema: dict, current_path_dict: dict,
+                  new_schema: dict) -> dict:
+        # Base case: if we reach a leaf in the path_dict, we stop.
+        if (not current_path_dict or not isinstance(current_schema, dict) or
+                not current_schema.get('properties')):
+            return current_schema
+
+        if 'properties' not in new_schema:
+            new_schema = {
+                key: current_schema[key]
+                for key in current_schema
+                if key != 'properties'
+            }
+            new_schema['properties'] = {}
+
+        for key, sub_schema in current_schema['properties'].items():
+            if key in current_path_dict:
+                # Recursively keep keys if further path dict exists
+                new_schema['properties'][key] = {}
+                new_schema['properties'][key] = keep_keys(
+                    sub_schema, current_path_dict[key],
+                    new_schema['properties'][key])
+
+        return new_schema
```

Comment:
Seems like we do not handle the `required` and `oneOf` field? Should we only keep those in the path_dict?

Reply:
Good point! Excluded those fields for now, as we don't need to support them.